client/extensions/woocommerce/app/promotions/helpers.js

```diff
@@ -0,0 +1,26 @@
+
+export function isValidPromotion( promotion ) {
+	if ( ! promotion || ! promotion.type ) {
+		return false;
+	}
+
+	const validCouponCode = promotion.couponCode && promotion.couponCode.length > 0;
+
```

Comment:
From a prior PR I commented on the end date of a promotion being able to be set before the start date... perhaps we could validate that here too eventually.

Reply:
I don't think I want to simply not light the Save button because start/end dates are tangled. I think that might be too confusing for the user.